core/src/main/scala/kafka/server/AbstractFetcherThread.scala

```diff
@@ -145,10 +145,9 @@ abstract class AbstractFetcherThread(name: String,
                 case Errors.NONE =>
                   try {
                     val messages = partitionData.toByteBufferMessageSet
+                    val validBytes = messages.validBytes
```

Comment:
Generally, we assign vals just before they need to be used so that you don't need to think about the consequences of such assignments until required.